stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/tcp/SelectorManager.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.socket.tcp;
+
+import org.ballerinalang.runtime.threadpool.BLangThreadFactory;
+import org.ballerinalang.stdlib.socket.exceptions.SelectorInitializeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousCloseException;
+import java.nio.channels.ClosedByInterruptException;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SelectableChannel;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import static java.nio.channels.SelectionKey.OP_READ;
+
+/**
+ * This will manage the Selector instance and handle the accept, read and write operations.
+ *
+ * @since 0.985.0
+ */
+public class SelectorManager {
+
+    private static final Logger log = LoggerFactory.getLogger(SelectorManager.class);
+
+    private Selector selector;
+    private boolean running = false;
+    private ThreadFactory threadFactory = new BLangThreadFactory("socket-selector");
+    private ExecutorService executor = Executors.newSingleThreadExecutor(threadFactory);
+    private boolean execution = true;
```

Comment:
Wondering if the boolean `execution` should be `executing` or something similar, to be consistent with the boolean `running`